pkg/scalers/etcd_scaler.go

```diff
@@ -0,0 +1,305 @@
+)
+
+type etcdScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *etcdMetadata
+	client     *clientv3.Client
+	logger     logr.Logger
+}
+
+type etcdMetadata struct {
+	endpoints                   []string
+	watchKey                    string
+	value                       float64
+	activationValue             float64
+	watchProgressNotifyInterval int
+	scalerIndex                 int
+	// TLS
+	enableTLS   bool
+	cert        string
+	key         string
+	keyPassword string
+	ca          string
+}
+
+// NewEtcdScaler creates a new etcdScaler
+func NewEtcdScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	meta, parseErr := parseEtcdMetadata(config)
+	if parseErr != nil {
+		return nil, fmt.Errorf("error parsing kubernetes workload metadata: %s", parseErr)
+	}
+
+	cli, err := getEtcdClients(meta)
+	if err != nil {
+		return nil, err
+	}
+	return &etcdScaler{
+		metricType: metricType,
+		metadata:   meta,
+		client:     cli,
+		logger:     InitializeLogger(config, "etcd_scaler"),
+	}, nil
+}
+
+func parseEtcdAuthParams(config *ScalerConfig, meta *etcdMetadata) error {
+	meta.enableTLS = false
+	if val, ok := config.AuthParams["tls"]; ok {
+		val = strings.TrimSpace(val)
+		if val == etcdTLSEnable {
+			certGiven := config.AuthParams["cert"] != ""
+			keyGiven := config.AuthParams["key"] != ""
+			if certGiven && !keyGiven {
+				return errors.New("key must be provided with cert")
+			}
+			if keyGiven && !certGiven {
+				return errors.New("cert must be provided with key")
+			}
+			meta.ca = config.AuthParams["ca"]
+			meta.cert = config.AuthParams["cert"]
+			meta.key = config.AuthParams["key"]
+			if value, found := config.AuthParams["keyPassword"]; found {
+				meta.keyPassword = value
+			} else {
+				meta.keyPassword = ""
+			}
+			meta.enableTLS = true
+		} else if val != etcdTLSDisable {
+			return fmt.Errorf("err incorrect value for TLS given: %s", val)
+		}
+	}
+
+	return nil
+}
+
+func parseEtcdMetadata(config *ScalerConfig) (*etcdMetadata, error) {
+	meta := &etcdMetadata{}
+	var err error
+	meta.endpoints = strings.Split(config.TriggerMetadata[endpoints], ",")
+	if len(meta.endpoints) == 0 || meta.endpoints[0] == "" {
+		return nil, fmt.Errorf("endpoints required")
+	}
+
+	meta.watchKey = config.TriggerMetadata[watchKey]
+	if len(meta.watchKey) == 0 {
+		return nil, fmt.Errorf("watchKey required")
+	}
+
+	meta.value, err = strconv.ParseFloat(config.TriggerMetadata[value], 64)
+	if err != nil || meta.value <= 0 {
+		return nil, fmt.Errorf("value must be a float greater than 0")
+	}
+
+	meta.activationValue = 0
+	if val, ok := config.TriggerMetadata[activationValue]; ok {
+		meta.activationValue, err = strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("activationValue must be a float")
+		}
+	}
+
+	meta.watchProgressNotifyInterval = 600
```

Comment:
Nit, could you please make constant for the `600`  value? ie `defaultWatchProgressNotifyInterval` ?

Reply:
done